packages/browser-wallet/src/shared/utils/token-helpers.ts

```diff
@@ -73,7 +73,12 @@ function deserializeTokenMetadataReturnValue(returnValue: string): string[] {
         const url = Buffer.from(buf.subarray(urlStart, urlEnd)).toString('utf8');
         urls.push(url);
 
-        cursor = urlEnd + 1;
+        if (buf[urlEnd] === 0) {
+            cursor = urlEnd + 1;
+        } else {
+            // Skip the metadata Hash.
+            cursor = urlEnd + 1 + 32;
+        }
```

Comment:
I don't understand what's happening here. If the last byte of the url bytes is 0, it doesn't include a hash? Also in the issue referenced, to me it sounds like the hash can be part of the path specifying the location of the metadata JSON, so skipping it won't actually solve the issue if that's the case. Could you explain where 32 comes from in terms of it being the amount needed to skip a hash in the url.

Maybe for some context, in single page application, router libraries sometimes use a router implementation called "hash router", which I believe is the case described in the issue.